pkg/base/base.go

```diff
@@ -102,129 +77,142 @@ func (e ParseError) Error() string {
+	return true, nil
+}
 
-			if strVal, ok := val.(string); ok {
-				boolVal, err := strconv.ParseBool(strVal)
-				if err != nil {
-					// should this be a ParseError?
-					return true, errors.Errorf("unable to parse %s as bool in when annotation of object %s, kind %s/%s", strVal, o.Metadata.Name, o.APIVersion, o.Kind)
-				}
+func (f BaseFile) GetKotsHookEvents() ([]HookEvent, error) {
+	obj, gvk, err := f.maybeDecode()
+	if err != nil || gvk == nil {
+		return nil, err
+	}
 
-				return boolVal, nil
-			}
+	annotations, _ := metadataAccessor.Annotations(obj)
 
-			// should this be a ParseError?
-			return true, errors.Errorf("unexpected type in when annotation of %s/%s: %T", o.APIVersion, o.Metadata.Name, val)
-		}
+	return getKotsHookEvents(annotations), nil
+}
+
+func (f BaseFile) IsKotsKind() (bool, error) {
+	_, gvk, err := f.maybeDecode()
+	if err != nil || gvk == nil {
+		return false, err
 	}
 
-	return true, nil
+	return isKotsAPIVersionKind(gvk), nil
 }
 
-func (f BaseFile) IsKotsKind() (bool, error) {
+func (f BaseFile) maybeDecode() (runtime.Object, *schema.GroupVersionKind, error) {
 	var m interface{}
 
 	if err := yaml.Unmarshal(f.Content, &m); err != nil {
 		// check if this is a yaml file
 		if ext := filepath.Ext(f.Path); ext == ".yaml" || ext == ".yml" {
-			return false, ParseError{Err: err}
+			return nil, nil, ParseError{Err: err}
 		}
-		return false, nil
+		return nil, nil, nil
 	}
 
-	o := OverlySimpleGVK{}
-	_ = yaml.Unmarshal(f.Content, &o) // error should be caught in previous unmarshal
-
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	obj, gvk, err := decode(f.Content, nil, nil)
 	// check if this is a kubernetes document
-	if o.APIVersion == "" || o.Kind == "" {
-		// check if this is a yaml file
+	if err != nil {
 		if ext := filepath.Ext(f.Path); ext == ".yaml" || ext == ".yml" {
 			// ignore empty files and files with only comments
 			if m == nil {
-				return false, nil
+				return nil, nil, nil
 			}
-			return false, ParseError{Err: errors.New("not a kubernetes document")}
+			return nil, nil, ParseError{Err: errors.New("not a kubernetes document")}
 		}
-		return false, nil
+		return nil, nil, nil
 	}
-
-	return iskotsAPIVersionKind(o), nil
+	return obj, gvk, err
 }
 
-func iskotsAPIVersionKind(o OverlySimpleGVK) bool {
-	if o.APIVersion == "velero.io/v1" && o.Kind == "Backup" {
+func isKotsAPIVersionKind(gvk *schema.GroupVersionKind) bool {
+	if gvk.Group == "velero.io" && gvk.Kind == "Backup" {
 		return true
 	}
-	if o.APIVersion == "kots.io/v1beta1" {
+	if gvk.Group == "kots.io" {
 		return true
 	}
-	if o.APIVersion == "troubleshoot.replicated.com/v1beta1" {
+	if gvk.Group == "troubleshoot.replicated.com" {
 		return true
 	}
 	// In addition to kotskinds, we exclude the application crd for now
-	if o.APIVersion == "app.k8s.io/v1beta1" {
+	if gvk.Group == "app.k8s.io" {
 		return true
 	}
 	return false
 }
 
+func isExcludedByAnnotation(annotations map[string]string) (bool, error) {
```

Comment:
this may break backwards compatibility for apps with annotations with bool values. i should probably test and fix this